src/profiling/profiling_enabled.rs

```diff
@@ -146,12 +145,29 @@ static CONTEXT_ID: AtomicU8 = AtomicU8::new(0);
 
 struct GpuCtx {
     id: u8,
-    query: [u32; 64 * 1024],
+    query: Vec<u32>,
     head: usize,
     tail: usize,
 }
 
 impl GpuCtx {
+    fn new() -> Self {
+        let len = 64 * 1024;
+        let mut query = Vec::with_capacity(len);
+        let remaining = query.spare_capacity_mut();
+        unsafe {
+            GenQueries(remaining.len() as i32, remaining.as_mut_ptr() as *mut u32);
+            query.set_len(len);
+        }
```

Comment:
I fixed a stack overflow that sometimes happens (depending on which optimizations the compiler does) with the statically sized array. It was first created on the stack and then moved to the struct.